### PR TITLE
Automatic prompting

### PR DESCRIPTION
**Pull request by Torantulino**
### Background

Prompt engineering, although in English, is a bad user experience that takes skill and practice.

This addition removes the need for laborious prompt engineering from Auto-GPT, by creating a system so easy that anybody can use it effectively.

Instead of having to manually enter the ai_config information, the user can simply now type something as simple as "Make pong", and a suitable ai_config will be automatically generated.

image

### Changes

+ Adds automatic ai_config generation as the default user flow.
+ Relegates manual ai_config creation to an optional "manual mode" 
+ Adds tests to cover automatic and manual ai_config creation
+ Displays AI details to user upon creation.

### Documentation

Docstrings and comments throughout

### Test Plan

I have implemented unit tests to cover the new and existing functions in the setup.py file.

The prompt system prompt will likely need to be further improved going forward so that it leans on AutoGPT's skill-set.

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Comment by richbeales** [comment]
linting fail

**Review by ntindle — APPROVED** [review]
add a note to news for this

**Comment by ntindle** [comment]
And fix the linting

**Review by ntindle — CHANGES_REQUESTED** [review]
also you mayyyyy have infinite tests so take a look at that

**Comment by Torantulino** [comment]
CI Tests should be fixed in #2904

File: autogpt/setup.py [hunk 1]
```diff
@@ -75,3 +134,64 @@ def prompt_user() -> AIConfig:
...
+        {
+            "role": "user",
+            "content": f"Task: '{user_prompt}'\nRespond only with the output in the exact format specified in the system prompt, with no explanation or conversation.\n",
+        },
+    ]
+    output = create_chat_completion(messages, CFG.fast_llm_model)
+
+    # Debug LLM Output
+    if CFG.debug_mode:
+        logger.debug(f"AI Config Generator Raw Output: {output}")
```
Comment (Pwuts) [inline autogpt/setup.py]: Assuming the logger is set up properly, the `if` isn't necessary and the logger will only output messages >= its configured log level. The `debug_mode` should ideally only be used to define that log level.

**Review by Pwuts — COMMENTED** [review]
This is definitely going to be good for UX. :)

File: autogpt/setup.py (hunk 1 above)
Comment (Torantulino) [reply autogpt/setup.py]: Ah, that's a great system if it is indeed set up that way, thanks for letting me know :)

**Comment by Torantulino** [comment]
Would be great to see some tests and prompt tweaks on the system prompt here.

It's not perfect yet, as you can see in the example it includes steps that don't play to the strengths of an LLM.

**Review by merwanehamadi — APPROVED** [review]

**Comment by Pwuts** [comment]
Why are the unit tests getting canceled?

**Review by richbeales — APPROVED** [review]

**Comment by ntindle** [comment]
@richbeales in tests/utils.py theres a new decorator to require an API key to exist before running the test that uses it. Can you add it to the tests as required here? You can see it used in https://github.com/Significant-Gravitas/Auto-GPT/blob/master/tests/integration/goal_oriented/test_write_file.py

**Review by ntindle — DISMISSED** [review]

**Review by richbeales — APPROVED** [review]

**Pull request merged**